core/pfe-core/controllers/scroll-spy-controller.ts

```diff
@@ -0,0 +1,166 @@
+import type { ReactiveController, ReactiveControllerHost } from 'lit';
+
+export interface ScrollSpyControllerOptions extends IntersectionObserverInit {
+  /**
+   * Tag names of legal link children.
+   * Legal children must have an `href` property/attribute pair, like `<a>`.
+   */
+  tagNames: string[];
+
+  /**
+   * Attribute to set on the active link element.
+   * @default 'active'
+   */
+  activeAttribute?: string;
+
+  /**
+   * The root node to query content for
+   * @default the host's root node
+   */
+  rootNode?: Node;
+}
+
+type HREFElement = Element & { href: string };
```

Comment:
TS question: There is an existing `HTMLAnchorElement` type.  Is there a benefit to going more generic with the `Element` type?

Reply:
Yes, we'd generally prefer to use the widest type possible here, since this controller should work with the most number of cases.

In our case specifically, pfe-jump-links-item does not implement the whole HTMLAnchorElement interface, but it does have an href property.

This is one of the assumptions I listed above - that link elements have an href property. One way to avoid this requirement would be to add some `getHash` option, which would default to `el => el.href`